### PR TITLE
Separate Actors reminders interface

### PR DESCRIPTION
**Pull request by DeepanshuA**
# Description

This is still a work in progress. Some UTs are still failing, I am fixing them.

## Issue reference

Please reference the issue this PR will close: #6591 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by yaron2** [comment]
@DeepanshuA what issue does this close?

**Comment by DeepanshuA** [comment]
> @DeepanshuA what issue does this close?

hi @yaron2, this is to target https://github.com/dapr/dapr/issues/6591.

**Comment by artursouza** [comment]
I am OK with this change as long as it does not bundle behavior changes and passes all our validations, including E2E tests, perf tests and SDK validation.

File: pkg/actors/config.go [hunk 1]
```diff
@@ -133,47 +107,40 @@ func NewConfig(opts ConfigOpts) Config {
...
-	return c.Reentrancy
-}
-
-func (c *Config) GetRemindersPartitionCountForType(actorType string) int {
-	if val, ok := c.EntityConfigs[actorType]; ok {
-		return val.RemindersStoragePartitions
-	}
-	return c.RemindersStoragePartitions
-}
+// func (c *Config) GetRemindersPartitionCountForType(actorType string) int {
```
Comment (yaron2) [inline pkg/actors/config.go:135]: Remove commented code

**Review by yaron2 — CHANGES_REQUESTED** [review]

File: pkg/actors/core/actor_lock.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package actors
+package core
```
Comment (yaron2) [inline pkg/actors/core/actor_lock.go:14]: It's recommended to keep packages as flat as possible unless and only separate if needed. I feel that `core` is not descriptive enough of what the package means. If there's a better package name then lets change to that, if not, I'd rather move that under `actors`.

**Comment by DeepanshuA** [comment]
Closing in lieu of https://github.com/dapr/dapr/pull/6669

**Pull request closed without merging**